packages/edit-post/src/components/admin-notices/index.js

```diff
@@ -0,0 +1,115 @@
+/**
+ * WordPress dependencies
+ */
+import { Component } from '@wordpress/element';
+import { withDispatch } from '@wordpress/data';
+
+/**
+ * Mapping of server-supported notice class names to an equivalent notices
+ * module status.
+ *
+ * @type {Map}
+ */
+const NOTICE_CLASS_STATUSES = {
+	'notice-success': 'success',
+	updated: 'success',
+	'notice-warning': 'warning',
+	'notice-error': 'error',
+	error: 'error',
+	'notice-info': 'info',
+};
+
+/**
+ * Returns an array of admin notice Elements.
+ *
+ * @return {Element[]} Admin notice elements.
+ */
+function getAdminNotices() {
+	// The order is reversed to match expectations of rendered order, since a
+	// NoticesList is itself rendered in reverse order (newest to oldest).
+	return [ ...document.querySelectorAll( '.notice' ) ].reverse();
```

Comment:
Don't we have a wrapper where to limit the fetched elements? it feels too easy to use a `.notice` in a plugin or something.

Reply:
> Don't we have a wrapper where to limit the fetched elements? it feels too easy to use a `.notice` in a plugin or something.

We could potentially limit it from the specific top-level wrapper element. I can't imagine there would be any notices passed from the server which wouldn't be intended to be interpreted into an equivalent "global" notices in the editor itself.